File: frontend/src/types/invoice.ts
export type InvoiceStatus =
  | "draft"
  | "generated"
  | "uploaded"
  | "pending"
  | "accepted"
  | "rejected"
  | "archived"

export interface InvoiceLine {
  id: string
  invoice_id: string
  line_number: number
  description: string
  quantity: number
  unit_code: string
  unit_price: number
  vat_rate: number
  vat_category_code: string
  line_total: number
}

export interface Invoice {
  id: string
  organization_id: string
  client_id: string
  invoice_number: string
  issue_date: string
  due_date: string
  currency: string
  status: InvoiceStatus
  invoice_type: "b2b" | "b2c" | "b2g"
  subtotal_amount: number
  vat_amount: number
  total_amount: number
  xml_content?: string
  anaf_upload_id?: string
  anaf_download_id?: string
  anaf_error_message?: string
  submitted_at?: string
  deadline_date?: string
  notes?: string
  created_at: string
  updated_at: string
  lines?: InvoiceLine[]
}

export interface CreateInvoiceDto {
  client_id: string
  issue_date: string
  due_date: string
  currency?: string
  invoice_type?: "b2b" | "b2c" | "b2g"
  notes?: string
  lines: Omit<InvoiceLine, "id" | "invoice_id">[]
}

File: backend/app/models/invoice.py
from pydantic import BaseModel
from typing import Optional
from datetime import date


class InvoiceLineCreate(BaseModel):
    line_number: int
    description: str
    quantity: float
    unit_code: str = "H87"
    unit_price: float
    vat_rate: float = 19.00
    vat_category_code: str = "S"
    line_total: float


class InvoiceCreate(BaseModel):
    client_id: str
    issue_date: date
    due_date: date
    currency: str = "RON"
    invoice_type: str = "b2b"
    notes: Optional[str] = None
    lines: list[InvoiceLineCreate]


class InvoiceResponse(BaseModel):
    id: str
    organization_id: str
    client_id: str
    invoice_number: str
    issue_date: date
    due_date: date
    currency: str
    status: str
    invoice_type: str
    subtotal_amount: float
    vat_amount: float
    total_amount: float
    anaf_error_message: Optional[str] = None
    deadline_date: Optional[date] = None
    notes: Optional[str] = None
    created_at: str
    updated_at: str
